src/Npgsql/NpgsqlConnector.cs

```diff
@@ -749,17 +742,11 @@ void Connect(NpgsqlTimeout timeout)
 
         async Task ConnectAsync(NpgsqlTimeout timeout, CancellationToken cancellationToken)
         {
-            EndPoint[] endpoints;
-            if (!string.IsNullOrEmpty(Host) && Host[0] == '/')
-            {
-                endpoints = new EndPoint[] { new UnixEndPoint(Path.Combine(Host, $".s.PGSQL.{Port}")) };
-            }
-            else
-            {
-                // Note that there aren't any timeout-able or cancellable DNS methods
-                endpoints = (await Dns.GetHostAddressesAsync(Host).WithCancellation(cancellationToken))
-                    .Select(a => new IPEndPoint(a, Port)).ToArray();
-            }
+            // Note that there aren't any timeout-able or cancellable DNS methods
+            var endpoints = Path.IsPathRooted(Host)
+                ? new EndPoint[] { new UnixEndPoint(Path.Combine(Host, $".s.PGSQL.{Port}")) }
+                : (await Dns.GetHostAddressesAsync(Host).WithCancellationAndTimeout(timeout, cancellationToken))
+                .Select(a => new IPEndPoint(a, Port)).ToArray();
```

Comment:
Does it makes sense to combine sync and async paths?

Reply:
We can consider it, though we do need to call different methods for Dns.GetHostAddresses, and the rest of the method is completely different anyway, so... I don't think it's worth it.